cmd/edgex-secrets-config/README.md

```diff
@@ -0,0 +1,179 @@
+
+    * **--client\_secret** (optional)
+
+      Optional manually-specified OAuth2 client_secret.  Will be generated if not present.  Equivalent to a password.
+
+    * **--redirect\_uris** _url\_for\_browser\_redirection_ (required for oauth2 tokens)
+
+      OAuth2 redirect URL for browser-based users.  Not currently used by EdgeX.
+
+    Upon completion, the command outputs a JSON structure containing the client_id and client_secret.
+    
+
+  * **deluser**
+
+    Delete a API gateway user. Requires additional arguments:
+
+    * **--user** _username_ (required)
+
+      Username of the user to delete.
+
+
+  * **jwt**
+
+    Utility function to create a JWT proxy authentication token from a supplied secret. This command does not require secret store access, but the values supplied must match those presented to the adduser command earlier. Requires additional arguments:
+
+    * **--algorithm** `RS256` | `ES256` (required)
+
+      Algorithm used for signing the JWT.
+      (See [RFC 7518](https://tools.ietf.org/html/rfc7518#section-3.1) for a list of signing algorithms.)
+
+    * **--id** _key_ (required)
+
+      The &quot;key&quot; field from the &quot;adduser&quot; command.
+      (This will be either the --id argument passed in, or the automatically generated identifier.)
+      (This is not actually a cryptographic key, but a unique identifier such as would be used in a database.)
+
+    * **--private\_key** _/path/to/private.key_ (required)
+
+      Private key used to sign the JWT (PEM-encoded) with a key type corresponding to the above-supplied algorithm.
+
+    * **--exp** _duration_ (optional)
+
+      Duration of generated JWT expressed as a golang-parseable duration value. Use &quot;never&quot; to omit an expiration field in the JWT. Defaults to &quot;1h&quot; (one hour) if unspecified.
+
+    
+    The generated JWT will be the encoded representation of:
+
+      <pre>
+      {
+        &quot;typ&quot;: &quot;JWT&quot;,
+        &quot;alg&quot;: &quot;RS256 | ES256&quot;
+      }
+      {
+        &quot;iss&quot;: &quot;_key_&quot;,
+        &quot;exp&quot;: (calculated expiration time)
+      }
+      (signature)
+      </pre>
+
+
+  * **oauth2**
+
+    Utility function to create an OAuth2 proxy authentication token using the client_credentials OAuth2 grant flow. This command does not require secret store access, but the values supplied must match those presented to the adduser command earlier. Requires additional arguments:
+
+    * **--client\_id** _client\_id_ (required)
+
+      Optional manually-specified OAuth2 client_id.  Will be generated if not present.  Equivalent to a username.
+
+    * **--client\_secret** _client\_secret_ (required)
+
+      Optional manually-specified OAuth2 client_secret.  Will be generated if not present.  Equivalent to a password.
+
+
+
+# CONFIGURATION
+
+# ENVIRONMENT
+
+  * **IKM\_HOOK**
```

Comment:
Is this still applicable? I didn't think access to Vault was required for any of these operations anymore?

Reply:
This feature will be needed when Colin implements the authentication password for the Kong admin API.  We will need to get the Kong admin password out of the secret store in order to call the Kong APIs.

Reply:
I was hoping we'd get to it this release cycle, but we didn't.